samza-kafka/src/main/scala/org/apache/samza/system/kafka/KafkaSystemAdmin.scala

```diff
@@ -148,6 +149,25 @@ class KafkaSystemAdmin(
 
   import KafkaSystemAdmin._
 
+  @volatile var adminClient: AdminClient = null
```

Comment:
AdminClient can also be used to create topic, delete topic, list all topics, get topic metadata, etc.

This is volatile because SystemAdmin.deleteRecords() and SystemAdmin.start() may potentially be called by different threads. Does this make sense?

Reply:
Thanks! Makes sense.